CI: Don't run slow KVM tests on every PR

mkroening opened the pull request:
It is still required via bors.

jounathaen commented:
Hmm, I'm a bit hesitant on this one.
This will definitely speed up things and make the whole process faster.
However, we will get faster machines for the KVM tests at sometime and more faster runners in parallel sounds like the better approach to me.
But we can use this solution until that point.

mkroening commented:
I added two more commits.

It is really time consuming to wait for CI right now. One PR submission causes 9 CI jobs (three jobs each on PR, on staging and on master) which are serially executed on our one runner. This really quickly builds up.

I changed the config to not run CI on master again, since that exact commit has already been tested on staging by bors.

Until we can do CI in parallel, this would make my life a lot easier.

jounathaen commented:
Well, we don't want that, do we?
bors r+

mkroening commented:
bors retry

mkroening commented:
The failure is not related to this PR, but to the newest nightly.

mkroening commented:
bors retry
